slycot/analysis.py

```diff
@@ -1628,7 +1630,118 @@ def ab13fd(n, A, tol = 0.0):
     raise_if_slycot_error(out[-1], arg_list, ab13fd.__doc__)
     return out[0], out[1]
 
+
+def ab13md(n, Z, nblock, itype, x=None):
+    """mubound, d, g, xout = ab13md(n, Z, nblock, itype, [x])
+
+    Find an upper bound for the structured singular value of complex
+    matrix Z and given block diagonal structure.
+
+    Parameters
+    ----------
+    n : integer
+      Order of Z; n=Z.shape[0].
+
+    Z : (n,n) complex array
+      Matrix to find structured singular value upper bound of
+
+    nblock : (m,) integer array
+      The size of the block diagonals of the uncertainty structure;
+      i.e., nblock(i)=p means that the ith block is pxp.
+
+    itype : (m,) integer array
+      The type of each block diagonal uncertainty defined in nblock.
+      itype(i)==1 means that the ith block is real, while itype(i)==2
+      means the the ith block is complex.  Real blocks must be 1x1,
+      i.e., if itype(i)==1, ntype(i) must be 1.
+
+    x : (q,) real array or None
+      If not None, must be the output of a previous call to ab13md.
+      The previous call must have been with the same values of n,
+      nblock, and itype; and the previous call's Z should be "close"
+      to the current call's Z.
+
+      q is determined by the block structure; see SLICOT AB13MD for
+      details.
+
+    Returns
+    -------
+    mubound : non-negative real scalar
+      Upper bound on structure singular value for given arguments
+
+    d, g : (n,) real arrays
+      Real arrays such that if D=np.diag(g), G=np.diag(G), and ZH = Z.T.conj(), then
+        ZH @ D**2 @ Z + 1j * (G@Z - ZH@G) - mu**2 * D**2
+      will be negative semi-definite.
+
+    xout : (q,) real array
+      For use as ``x`` argument in subsequent call to ``ab13md``.
+
+    For scalar Z and real uncertainty (ntype=1, itype=1), returns 0
+    instead of abs(Z).
+
+    Raises
+    ------
+    SlycotArithmeticError
+        :info = 1: Block sizes must be positive
+        :info = 2: Block sizes must sum to n
+        :info = 3: Real blocks must be of size 1
+        :info = 4: Block types must be 1 or 2
+        :info = 5: Error in linear equation solution
+        :info = 6: Error in eigenvalue or singular value computation
+
+    Notes
+    -----
+    This wraps SLICOT routine AB13MD, which implements the upper bound
+    of [1].
+
+    References
+    ----------
+    .. [1] Fan, M.K.H., Tits, A.L., and Doyle, J.C., "Robustness in
+       the presence of mixed parametric uncertainty and unmodeled
+       dynamics," IEEE Trans. Automatic Control, vol. AC-36, 1991,
+       pp. 25-38.
+
+    """
+    hidden = ' (hidden by the wrapper)'
+    arg_list = ['fact' + hidden, 'n' + hidden, 'z', 'ldz' + hidden,
+                'm' + hidden, 'nblock', 'itype', 'x', 'bound', 'd', 'g',
```

Comment:
`n` is not hidden

Reply:
So "hidden" here is not arguments hidden from the caller of slycot.ab13md?

Reply:
Those are the arguments of the Fortran subroutine hidden in the Python call signature of `_wrapper.ab13md()`. The `arg_list` is used for the error message handler which references the argument count and order of the Fortran routine.

Reply:
OK, will fix.

Reply:
Hm, what about `m` and `info`? `m` is apparently optional, despite my attempts to make it required in analysis.pyf, and `info` is hidden in `ab13ed`.

Reply:
Not sure about `m`. Maybe f2py is confused by the `dimension(m)`  and `depend(m)` stuff.

`info` shouldn't be hidden. It is in the output and you need it for the error handler.